client/src/utils.ts

```diff
@@ -166,3 +166,27 @@ export function launchLanguageServer(context: vscode.ExtensionContext): Language
 
   return client
 }
+
+export function getErrorDiagnosticsBySource(): Map<string, Array<[vscode.Uri, vscode.Diagnostic]>> {
+  const errorsBySource = new Map<string, Array<[vscode.Uri, vscode.Diagnostic]>>()
+  const diagnostics = vscode.languages.getDiagnostics()
+  let uri, fileDiagnostics
+
+  for ([uri, fileDiagnostics] of diagnostics) {
```

Comment:
Subjective: Is this syntax possible here, so we don't need to re-bind the same variables every loop?
```suggestion
  for (const [uri, fileDiagnostics] of diagnostics) {
```